user-service/src/main/java/com/innowise/userservice/controller/UserController.java

```diff
@@ -0,0 +1,112 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+/**
+ * Controller for managing user-related operations.
+ * Provides REST endpoints for creating, retrieving, updating, and deleting user information.
+ */
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/users")
+public class UserController {
+
+    private final UserService userService;
+
+    /**
+     * Creates a new user.
+     *
+     * @param userRequest The {@link UserRequest} object containing the data for the new user.
+     * @return A {@link ResponseEntity} with the created {@link UserResponse} object and an HTTP status of OK (200).
+     */
+    @PostMapping
+    public ResponseEntity<UserResponse> save(@RequestBody @Valid UserRequest userRequest) {
+        UserResponse user = userService.save(userRequest);
+
+        return ResponseEntity.ok(user);
+    }
+
+    /**
+     * Finds a user by their unique ID and returns it along with their associated card information.
+     *
+     * @param id The unique ID of the user.
+     * @return A {@link ResponseEntity} with the {@link UserResponse} object and an HTTP status of OK (200).
+     */
+    @GetMapping("/{id}")
+    public ResponseEntity<UserResponse> findById(@PathVariable("id") Long id) {
+        UserResponse user = userService.findById(id);
+
+        return ResponseEntity.ok(user);
+    }
+
+    /**
+     * Retrieves a list of users based on the specified filtering criteria.
+     *
+     * @param filter A string that determines the filtering logic to be applied.
+     * Defaults to "pageable" if not provided.
+     * @param ids A list of user IDs to filter by, used when {@code filter="ids"}.
+     * Optional.
+     * @param email The email address of the user to retrieve, used when {@code filter="email"}.
+     * Optional.
+     * @param pageable Pagination and sorting information, used when {@code filter="pageable"}.
+     * @return A {@link ResponseEntity} containing a {@link List} of {@link UserResponse} DTOs and an HTTP status of OK (200).
+     */
+    @GetMapping
+    public ResponseEntity<List<UserResponse>> findByFilter(
+            @RequestParam(required = false, defaultValue = "pageable") String filter,
+            @RequestParam(required = false) List<Long> ids,
+            @RequestParam(required = false) String email,
+            Pageable pageable) {
```

Comment:
No max page size limit
Security Risk: User could request page size of 1,000,000
Recommendation: Add @PageableDefault(size = 20, page = 0) and max size validation